utils/utils_hash.c

```diff
@@ -43,7 +42,47 @@ int hash_ctx_copy( hash_ctx *nctx, const hash_ctx *octx ) {
 
 int hash_final_digest( unsigned char *out, size_t outlen, hash_ctx *ctx ) {
     int ok = EVP_DigestFinalXOF(ctx->x, out, outlen);
-    EVP_MD_CTX_destroy(ctx->x);
+    EVP_MD_CTX_free(ctx->x);
+    return (ok) ? 0 : -1;
+}
+
+#elif (defined(_UTILS_OPENSSL_)||defined(_UTILS_SUPERCOP_)) && defined(_UTILS_OQS_)
```

Comment:
Something is off here. 
There is a `#if defined (_UTILS_OQS_)` further down in the file. That is the branch that should be taken.